pkg/image/importer/importer.go

```diff
@@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
 	}
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
+	bs := repo.Blobs(ctx)
+
+	layerSet := sets.NewString()
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
According to the v2 spec size must be filled out in layers.  Is that not
the case?

On Wed, Jun 1, 2016 at 2:09 PM, Alexey Gladkov <notifications@github.com>
wrote:

> In pkg/image/importer/importer.go
> <https://github.com/openshift/origin/pull/9115#discussion_r65413168>:
>
> > @@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
> >  	}
> >  }
> >
> > +func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
> > +	bs := repo.Blobs(ctx)
> > +
> > +	layerSet := sets.NewString()
> > +	size := int64(0)
> > +	for i := range image.DockerImageLayers {
> > +		layer := &image.DockerImageLayers[i]
> > +		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
>
> schema1ToImage is only about image creation. It's not getting digests and
> do not have anything to calculate the size.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9115/files/492baa03f23cf45ca229c3501c29ca4bccf546f3#r65413168>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p8C7_D83OxQ7joXFIweSMM4eSYOUks5qHcrygaJpZM4IrzC2>
> .
>